results/cmd/api/main_test.go

```diff
@@ -114,14 +115,15 @@ func TestUpdateTaskRun(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 	objectmeta := &pb.ObjectMeta{Uid: "123459", Name: "mytaskrun", Namespace: "default"}
-	r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "1", Metadata: objectmeta}})
+	r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1beta1", Metadata: objectmeta}})
 	if err != nil {
 		t.Fatalf("could not create taskrun: %v", err)
 	}
 
 	// Update the created taskrun
 	objectmeta = &pb.ObjectMeta{Uid: "123459", Name: "ziweifan", Namespace: "tomorrow"}
-	r, err = srv.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1alpha1", Metadata: objectmeta}, ResultsId: r.GetResultsId()})
+	fmask, _ := fieldmask.New(&pb.TaskRun{}, "api_version", "metadata.uid")
```

Comment:
There's a few different things we should test here:

1. No mask
2. Partial update with mask
   - Verify only fields that should be modified are, and that fields that were not included stay the same.
3. Mask with excess fields - e.g. setting a field in the mask but not setting a value in the request (this should unset the value).
4. Mask with empty fields.
5. Mask with invalid fields.
6. Interaction with repeated fields (see https://google.aip.dev/144#update-strategies)
6. Any more test cases you can think of.

I recommend representing this as a table driven test - https://github.com/golang/go/wiki/TableDrivenTests

Reply:
done